weed/shell/command_volume_fsck.go

```diff
@@ -210,66 +231,68 @@ func (c *commandVolumeFsck) collectFilerFileIdAndPaths(volumeIdToServer map[uint
-		totalInUseCount += inUseCount
-		totalOrphanChunkCount += uint64(len(orphanFileIds))
-		totalOrphanDataSize += orphanDataSize
-
-		if verbose {
-			for _, fid := range orphanFileIds {
-				fmt.Fprintf(writer, "%s\n", fid)
+	for dataNodeId, volumeIdToVInfo := range dataNodeVolumeIdToVInfo {
+		for volumeId, vinfo := range volumeIdToVInfo {
+			inUseCount, orphanFileIds, orphanDataSize, checkErr := c.oneVolumeFileIdsSubtractFilerFileIds(tempFolder, dataNodeId, volumeId, writer, verbose)
+			if checkErr != nil {
+				return fmt.Errorf("failed to collect file ids from volume %d on %s: %v", volumeId, vinfo.server, checkErr)
 			}
-		}
+			totalInUseCount += inUseCount
+			totalOrphanChunkCount += uint64(len(orphanFileIds))
+			totalOrphanDataSize += orphanDataSize
 
-		if applyPurging && len(orphanFileIds) > 0 {
 			if verbose {
-				fmt.Fprintf(writer, "purging process for volume %d", volumeId)
-			}
-
-			if vinfo.isEcVolume {
-				fmt.Fprintf(writer, "skip purging for Erasure Coded volume %d.\n", volumeId)
-				continue
+				for _, fid := range orphanFileIds {
+					fmt.Fprintf(writer, "%s\n", fid)
+				}
 			}
 
-			needleVID := needle.VolumeId(volumeId)
+			if applyPurging && len(orphanFileIds) > 0 {
+				if verbose {
+					fmt.Fprintf(writer, "purging process for volume %d", volumeId)
+				}
 
-			if vinfo.isReadOnly {
-				err := markVolumeWritable(c.env.option.GrpcDialOption, needleVID, vinfo.server, true)
-				if err != nil {
-					return fmt.Errorf("mark volume %d read/write: %v", volumeId, err)
+				if vinfo.isEcVolume {
+					fmt.Fprintf(writer, "skip purging for Erasure Coded volume %d.\n", volumeId)
+					continue
 				}
 
-				fmt.Fprintf(writer, "temporarily marked %d on server %v writable for forced purge\n", volumeId, vinfo.server)
-				defer markVolumeWritable(c.env.option.GrpcDialOption, needleVID, vinfo.server, false)
-			}
+				needleVID := needle.VolumeId(volumeId)
 
-			fmt.Fprintf(writer, "marked %d on server %v writable for forced purge\n", volumeId, vinfo.server)
+				if vinfo.isReadOnly {
+					err := markVolumeWritable(c.env.option.GrpcDialOption, needleVID, vinfo.server, true)
+					if err != nil {
+						return fmt.Errorf("mark volume %d read/write: %v", volumeId, err)
+					}
 
-			if verbose {
-				fmt.Fprintf(writer, "purging files from volume %d\n", volumeId)
-			}
+					fmt.Fprintf(writer, "temporarily marked %d on server %v writable for forced purge\n", volumeId, vinfo.server)
+					defer markVolumeWritable(c.env.option.GrpcDialOption, needleVID, vinfo.server, false)
+				}
+
+				fmt.Fprintf(writer, "marked %d on server %v writable for forced purge\n", volumeId, vinfo.server)
+
+				if verbose {
+					fmt.Fprintf(writer, "purging files from volume %d\n", volumeId)
+				}
 
-			if err := c.purgeFileIdsForOneVolume(volumeId, orphanFileIds, writer); err != nil {
-				return fmt.Errorf("purging volume %d: %v", volumeId, err)
+				if err := c.purgeFileIdsForOneVolume(volumeId, orphanFileIds, writer); err != nil {
```

Comment:
This also will repeatedly processing one volumeId.

I wonder how to test the change properly.

Reply:
> I wonder how to test the change properly.

done